### PR TITLE
Remove redundant line, warn user if invalid credentials, add convert_to_oauth_credentials function

### PR DESCRIPTION
**Pull request by jasonqng**
This PR does 3 things: 

1. Removes a redundant line in utils.py 
2. Prints a warning to console to the user if they supply credentials and they are invalid. This is important because at the moment, if a user supplies credentials (say a collective team account's creds) and it happens to be invalid, it silently loads from `~/.oauth/drive.json`, which may have unintended consequences and/or give the user the false impression that the supplied credentials are working.
3. Adds a helper function that converts a user-supplied token file to an oauth object which can be passed directly to the `upload` or `download` functions as the `credentials` param.

To test the new helper function:
Checkout branch and from the `df2gspread/` folder (the one with `/bin`, `/docs`, etc), in python shell:

```
from df2gspread import df2gspread as d2g
from df2gspread import utils as utils
import pandas as pd
import numpy as np
# Copy the default token to an arbitrary location if you have one. Otherwise, skip
# and change the path in the convert_to_oauth_credentials() line to wherever the file is.
cp ~/.oauth/drive.json /tmp/drive.json
credentials = utils.convert_to_oauth_credentials('/tmp/drive.json')

df = pd.DataFrame(data = np.array([np.arange(100)]*1).T)
#Change to your spreadsheet
spreadsheet = '1EfKM7Yw56...'
wks_name = 'New Sheet123'
d2g.upload(df, spreadsheet, wks_name, credentials=credentials)
```
Should upload successfully even if you delete or don't have `~/.oauth/drive.json`. Can confirm by repeating the above after renaming or deleting `~/.oauth/drive.json`.

Now change the last line to gibberish like:
```
d2g.upload(df, spreadsheet, wks_name, credentials='fjashfads')
```

If you do this in master, if you have a valid `~/.oauth/drive.json`, it'll still upload with no warning message. In this branch, it'll now print to screen before successfully uploading using `~/.oauth/drive.json`:

```
>>> d2g.upload(df, spreadsheet, wks_name, credentials='fjashfads')
Invalid credentials supplied. Will generate from default token.
```

### CODE REVIEW

**Comment by jasonqng** [comment]
Updated the testing instructions in description.

**Comment by jasonqng** [comment]
Tests:

```
➜  df2gspread git:(tweaks) ✗ py.test tests/ -v
===================================== test session starts =====================================
platform darwin -- Python 2.7.10, pytest-2.9.2, py-1.4.31, pluggy-0.3.1 -- /usr/bin/python
cachedir: .cache
rootdir: /Users/jpersonal/Projects/df2gspread, inifile:
collected 8 items

tests/test_credentials.py::test_client_secrets_credentials xfail
tests/test_df2gspread.py::test_global_import PASSED
tests/test_df2gspread.py::test_version_check PASSED
tests/test_df2gspread.py::test_spreadsheet_invalid_file_id PASSED
tests/test_df2gspread.py::test_worksheet_invalid_name PASSED
tests/test_df2gspread.py::test_worksheet PASSED
tests/test_df2gspread.py::test_big_worksheet PASSED
tests/test_df2gspread.py::test_delete_file PASSED

============================ 7 passed, 1 xfailed in 95.07 seconds =============================
```

File: df2gspread/utils.py [hunk 1]
```diff
@@ -50,6 +48,13 @@ def run(cmd):
         sys.exit(process.returncode)
     return output, errors
 
+def convert_to_oauth_credentials(token_file=None):
```
Comment (maybelinot) [inline df2gspread/utils.py]: Can you also change modify `get_credentials()` function which has almost the same part of code (DRY)?
Otherwise LGTM!

Comment (jasonqng) [inline df2gspread/utils.py]: Absolutely, great point.

**Comment by maybelinot** [comment]
@jasonqng I made some changes to this PR and pushed in it master..

**Pull request closed without merging**